hypothesis-python/src/hypothesis/internal/conjecture/engine.py

```diff
@@ -1771,6 +1771,32 @@ def buffer(self):
     def blocks(self):
         return self.shrink_target.blocks
 
+    def all_block_bounds(self):
+        return self.shrink_target.all_block_bounds()
```

Comment:
I inspected all of the callers, and I'm pretty sure that none of them will update the target while iterating over this list.

(If they did, then they would have been wrong before I touched them, since they were previously iterating directly over `blocks`.)

The places that do need up-to-date information should already all be doing the right thing.